services/common/serv_helper.js

```diff
@@ -5,11 +5,111 @@
+    function aclPathFromURL(aUrl) {
+      if (!aUrl) {
+        debug('Trying to recover acl file from no url');
+        return {};
+      }
+      // Error ctrl?
+      var url = new URL(aUrl);
+      var pathname = url.pathname;
+      return pathname.substring(0, pathname.lastIndexOf('/') + 1);
+    }
+    return new Promise((resolve, reject) => {
+      var xhr = new XMLHttpRequest();
+      xhr.open('GET', aclPathFromURL(aUrl) + ACCESS_CTRL_FILE, true);
+      xhr.responseType = 'json';
+
+      xhr.onerror = function(error) {
+        reject(error);
+      };
+      xhr.onload = function() {
+        if (xhr.response !== null) {
+          var response = xhr.response;
+          for (var developer in response) {
+            var allowedOrgs = response[developer].allowedOrigins;
+            if (!allowedOrgs) {
+              reject(new Error('Wrong acl file. Developer ' + developer +
+                     ' does not have allowedOrigins keys'));
+            }
+            var allowedOrgsRE = [];
+            allowedOrgs.forEach(org => allowedOrgsRE.push(new RegExp(org)));
+            response[developer].allowedOrigins = allowedOrgsRE;
+          }
+          resolve(response);
+        } else {
+          reject(new Error('No valid JSON object was found (' +
+			    xhr.status + ' ' + xhr.statusText + ')'));
+        }
+      };
+      xhr.send();
+    });
+  };
+
+  function getRules(aAcl, aUrl) {
+    for (var developer in aAcl) {
+      var allowedOrgs  = aAcl[developer].allowedOrigins;
+      if (allowedOrgs.some(regexp => regexp.test(aUrl))) {
+        return aAcl[developer].rules;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Return true if:
+   *  - aAcl or aTargetURL or aData are not received
+   *  - No allowed Origins or rules has not been defined for allowed origin
+   *  - Constrainst has not been defined for an expecified operation (e.i.
+   *    operation is not included in rules)
+   *  - operation contain settings and not all of them are in constraint
+   * Return false if:
+   *  - Match with an allowed origin and
+   *    - rules is defined with no constraints (i.e. rules: {}) or
+   *    - constraints for an operation is empty (i.e. "operation":[]) or
+   *    - constraint contain settings and all settings required are in it
+   *
+   */
+  function isForbidden(aAcl, aTargetURL, aOperation, aValidator) {
+    if (!aAcl || !aTargetURL || !aOperation || !aValidator ||
+        (typeof aValidator !== 'function')) {
+      return true;
+    }
+    var rules = getRules(aAcl, aTargetURL);
+    // I'm so sorry, but you need to explicitly allow everything.
+    // If you want everybody to access everything write:
+    // {
+    //   "allowedEverybody": {
+    //     "allowedOrigins": [".*"],
+    //     "rules": {}
+    //   }
+    // }
+    if (!rules) {
+      return true;
+    }
+
+    if (Object.keys(rules).length === 0) {
+      return false;
+    }
+    var constraints = rules[aOperation];
+
```

Comment:
return !constraints || (constraints.length != 0 && aValidator(constraints));
(and remove the following if)